format/Flight.proto

```diff
@@ -525,3 +525,108 @@ message FlightData {
 message PutResult {
   bytes app_metadata = 1;
 }
+
+/*
+ * EXPERIMENTAL: Union of possible value types for a Session Option to be set to.
+ */
+message SessionOptionValue {
+  message StringListValue {
+    repeated string values = 1;
+  }
+
+  oneof option_value {
+    string string_value = 1;
+    bool bool_value = 2;
+    sfixed32 int32_value = 3;
+    sfixed64 int64_value = 4;
+    float float_value = 5;
+    double double_value = 6;
+    StringListValue string_list_value = 7;
+  }
+}
+
+/*
+ * EXPERIMENTAL: A request to set session options for an existing or new (implicit)
+ * server session.
+ *
+ * Sessions are persisted and referenced via RFC 6265 HTTP cookies, canonically
+ * 'arrow_flight_session_id', although implementations may freely choose their own name.
+ */
+message SetSessionOptionsRequest {
+  map<string, SessionOptionValue> session_options = 1;
+}
+
+/*
+ * EXPERIMENTAL: The results (individually) of setting a set of session options.
+ */
+message SetSessionOptionsResult {
+  enum Status {
+    // The status of setting the option is unknown. Servers should avoid using
+    // this value (send a NOT_FOUND error if the requested query is
+    // not known). Clients can retry the request.
+    SET_SESSION_OPTION_RESULT_UNSPECIFIED = 0;
```

Comment:
Why this?

Reply:
This is Protobuf convention, as stated before.

Reply:
Hmm, I'm curious about this. The [protobuf language guide](https://protobuf.dev/programming-guides/proto3/#enum) does state that "there must be a zero value, so that we can use 0 as a numeric default value". It doesn't state what this default value _should mean_, though.

Since this enum denotes an error return, it would be reasonable for the default value to mean success, rather than "unspecified".

Reply:
In any case, it seems weird to specify a default value that we explicit discourage servers from sending...

Reply:
https://protobuf.dev/programming-guides/style/#enums
https://buf.build/docs/best-practices/style-guide#enums
https://github.com/uber/prototool/blob/dev/style/README.md#enums

Reply:
Ok, so can we be clearer in the comments? Right now they give the impression that this is a meaningful value.
```suggestion
    // Default value for this enum. Server MUST NOT send this value explicitly.
    // Clients MAY receive this value if the server has sent an enum value that
    // is outside the range of values known by the client (for example if the server
    // conforms to a later version of this spec).
    SET_SESSION_OPTION_RESULT_UNSPECIFIED = 0;
```

Reply:
While it's a valid point to make the documentation of these values more clear, I believe it needs to be done globally to avoid implying differences in the different enum values' meaning if we document them differently.

Reply:
This is true. Perhaps we can find a terse convention that would point to a common explanation somewhere at the top of the file?

Reply:
I'm inclined to do the canonical thing here which is to simply reference https://protobuf.dev/programming-guides/style/#enums in the code (again maybe in a single place as you suggest, w/ "See note <at top or similar>" in the immediate comment), so people wanting to care about why Proto does what it does can read the thing without further ado.